src/js/components/CheckBox/stories/Temp.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+
+import { Box, Grommet, CheckBox, FormField, DataTable } from 'grommet';
+// import { grommet } from 'grommet/themes';
+import { deepMerge } from 'grommet/utils';
+import { hpe } from 'grommet-theme-hpe';
+
+// Source code for the data can be found here
+// https://github.com/grommet/grommet/blob/master/src/js/components/DataTable/stories/data.js
+import { columns, DATA } from '../../DataTable/stories/data';
+
+const controlledColumns = columns.map(col => ({ ...col }));
+delete controlledColumns[0].footer;
+delete controlledColumns[3].footer;
+delete controlledColumns[4].footer;
+delete controlledColumns[4].aggregate;
+
+const myTheme = deepMerge(hpe, {
+  checkBox: {
+    check: {
+      extend: props =>
+        props.inFormField ? `box-shadow: none` : `box-shadow: undefined`,
```

Comment:
Maybe I got lost, but should this be check for `componentContext` value of "FormField" in this approach?

Reply:
You are correct. Fixed.

Reply:
Thanks. Another question on this approach. Is ComponentContext something that would just live internally, or is it a Context, like responsive/theme contexts that we'd exposed to users? For example, I want my buttons in my navigation bar to be styled like X but want them to be styled in my main "blog" area as Y.

Is that something we want to introduce or is does that create too many ways to theme? (posing this as an open-ended question)

Reply:
I think it has the potential to serve externally and be consumed similarly to ResponsiveContext. With ResponsiveContext, I choose to affect prop values, layout, or styling based on screen width. Similarly, I could see ComponentContext allowing me to affect prop values, layout, and styling based on its container. I think your navigation vs. "blog" is a good example. Another example might be if I have a Notifications component that I'd like to re-use throughout my app. That notification behavior may vary based on where the notification is surfaced.

In a way, TableCell is applying a similar concept; when I am in a TableHeader context, do X; when I am in a TableBody context, do Y ... . Different implementation, but similar in concept that my component's properties may change based on their container.

That said, the direction should be informed and driven by concrete use cases rather than speculative.

Reply:
Another use case would be situations where I may want more granular control to alter my component based on its container's dimensions. For example, ResponsiveContext lets me do that based on window dimension, however sometimes I want my component to behave differently when contained in a small Box vs. a medium Box.